SMLHelper/Patchers/CraftDataPatcher.cs

```diff
@@ -16,12 +13,21 @@ public class CraftDataPatcher
         public static Dictionary<TechType, HarvestType> customHarvestTypeList = new Dictionary<TechType, HarvestType>();
         public static Dictionary<TechType, Vector2int> customItemSizes = new Dictionary<TechType, Vector2int>();
         public static Dictionary<TechType, EquipmentType> customEquipmentTypes = new Dictionary<TechType, EquipmentType>();
-        public static Dictionary<TechGroup, Dictionary<TechCategory, List<TechType>>> customGroups = new Dictionary<TechGroup, Dictionary<TechCategory, List<TechType>>>();
+        private static Dictionary<TechGroup, Dictionary<TechCategory, List<TechType>>> customGroups = new Dictionary<TechGroup, Dictionary<TechCategory, List<TechType>>>();
 
         public static List<TechType> customBuildables = new List<TechType>();
 
         private static readonly Type CraftDataType = typeof(CraftData);
 
+        public static void AddToCustomGroup(TechGroup group, TechCategory category, TechType techType)
```

Comment:
I think this would be easier to use. I don't know if it's necacarry to make customGroups private, but the idea is simpple - don't touch what doesn't need to be touched =)
And new usage example (once again - from your BaseLightSwitch)

            CraftDataPatcher.AddToCustomGroup(TechGroup.InteriorModules, TechCategory.InteriorModule,techType);

Way easier, if you'll ask me =(^^)=